ColorPicker/ColorPicker/ColorPicker.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  ColorPicker.swift
+//  ColorPicker
+//
+//  Created by Daria Tokareva on 22.06.2021.
+//
+
+import UIKit
+
+protocol ColorPickerDelegate {
+    func colorPicker(colorPicker: ColorPicker, hueValue: CGFloat)
+}
+
+class ColorPicker: UIView {
+
+    var gradientView: UIView!
+    var gradientLayer = CAGradientLayer()
+    var thumbView: UIView!
+    var chosenColor: UIColor = UIColor(hue: 0, saturation: 1, brightness: 1, alpha: 1)
+    var currentHue: CGFloat?
+    let hueColors = stride(from: 0, to: 1, by: 0.03).map {
+        UIColor(hue: $0, saturation: 1, brightness: 1, alpha: 1).cgColor
+    }
+    var delegate: ColorPickerDelegate?
```

Comment:
Для избежания утечек памяти ссылку на делегат необходимо сделать слабой (weak) 
```suggestion
    weak var delegate: ColorPickerDelegate?
```
При этом тип, который реализует протокол ColorPickerDelegate должен быть классом (ограничить протокол с помощью AnyObject или UIViewController).